File: Javascript/83.remove-duplicates-from-sorted-list.js
/*
 * @lc app=leetcode id=83 lang=javascript
 *
 * [83] Remove Duplicates from Sorted List
 */

function ListNode(val, next) {
     this.val = (val===undefined ? 0 : val)
     this.next = (next===undefined ? null : next)
}

// @lc code=start
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var deleteDuplicates = function(head) {
    if (head === null) {
        return null;
    }

    const node = new ListNode(head.val);
    let currentNode = node;
    let checkTargetNode = head.next;

    while (checkTargetNode !== null) {
        if (currentNode.val === checkTargetNode.val) {
            checkTargetNode = checkTargetNode.next;
            continue;
        }

        currentNode.next = new ListNode(checkTargetNode.val);
        currentNode = currentNode.next;
        checkTargetNode = checkTargetNode.next;
    }

    return node;
};
// @lc code=end

File: Ruby/83.remove-duplicates-from-sorted-list.rb
#
# @lc app=leetcode id=83 lang=ruby
#
# [83] Remove Duplicates from Sorted List
#

# @lc code=start
# Definition for singly-linked list.
class ListNode
    attr_accessor :val, :next
    def initialize(val = 0, _next = nil)
        @val = val
        @next = _next
    end
end
# @param {ListNode} head
# @return {ListNode}
def delete_duplicates(head)
    current = head
    while current != nil && current.next != nil
        if current.val != current.next.val
            current = current.next
            next
        end

        current.next = current.next.next
    end

    return head
end
# @lc code=end
